sympy/ntheory/factor_.py

```diff
@@ -1429,6 +1429,13 @@ def divisor_count(n, modulus=1):
         return 0
     return Mul(*[v + 1 for k, v in factorint(n).items() if k > 1])
 
+def proper_divisors(n, generator = False):
+    return divisors(n)[0:-1]
```

Comment:
This will not respect the generator setting, will it?
```python
>>> (i for i in range(3))[:-1]
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: 'generator' object has no attribute '__getitem__'
```

See #15928 which only needs all the changes condensed into a single commit.

Reply:
I though that it should take generator setting. But it is not taking generator setting while computing proper divisors.

Reply:
@smichr What is the need of returning a generator when generator=True? Can't we return unordered list only. A pseudo code can be, assuming  _divisors will return generator - 
```suggestion
    def proper_divisors(n, generator = False):
    if not generator:
        return divisors(n,generator)[0:-1]
    else:
        d = divisors(n,generator)
        c = []
        for i in d:
            if i != n:
                c.append(i)
        return c
```

Reply:
> What is the need of returning a generator when generator=True

because that is how the function currently works; there might be a LOT of divisors and one might not want to actually look at all of them -- or need them in a list -- so the generator option allows them to be retrieved one at a time.